### PR TITLE
Adding oc_label to lib_openshift.

### PR DESCRIPTION
**Pull request by kwoodson**

### CODE REVIEW

File: roles/lib_openshift/src/doc/label [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+DOCUMENTATION = '''
+---
+module: oc_label
+short_description: Create, modify, and idempotently manage openshift labels.
+description:
+  - Modify openshift labels programmatically.
+options:
+  state:
+    description:
+    - State represents whether to create, modify, delete, or list
```
Comment (tbielawa) [inline roles/lib_openshift/src/doc/label]: > State represents whether to create, modify, delete, or list

I see the choices below are ` ["present", "absent", "list", "add"]` so then:

* `create` maps to `add`
* `delete` maps to `absent`
* `list` maps to `list` 
* `modify` maps to `present`?


Maybe like this?

```yaml
    description:
    - State controls what to do with the module
    - An new label is added with the `create` state
    - An existing label is removed (or verified as removed) with the `absent` state
    - Applied labels may be enumerated with the `list` state
    - A specific labels existence is determined with the `present` state
```

But I'm not sure which of those states accounts for modifying objects

File: roles/lib_openshift/src/doc/label [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    required: false
+    default: /etc/origin/master/admin.kubeconfig
+    aliases: []
+  debug:
+    description:
+    - Turn on debug output.
+    required: false
+    default: False
+    aliases: []
+  kind:
```
Comment (tbielawa) [inline roles/lib_openshift/src/doc/label]: The `AnsibleModule` spec for `kind` defines it as such:

```python
    kind=dict(
        default=None,
        type='str',
        required=True,
        choices=['node', 'pod', 'namespace']
),
```

However the `DOCUMENTATION` does not match

Param |`AnsibleModule` spec | `DOCUMENTATION`
------------ | ------------ | -------------
default | `None` | **node**
required | `True` | ⁿ/ₐ

File: roles/lib_openshift/src/doc/label [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    - The kind of object that can be managed.
+    default: node
+    choices:
+    - node
+    - pod
+    - namespace
+    aliases: []
+  labels:
+    description:
+    - A list of labels to for the resource.
```
Comment (tbielawa) [inline roles/lib_openshift/src/doc/label]: A list of labels to apply to the resource. Each item in the list is a hash (ex: `region: east`)

File: roles/lib_openshift/src/doc/label [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+EXAMPLES = '''
+- name: Add label to node
+  oc_label:
+    name: ip-172-31-5-23.ec2.internal
+    state: add
+    kind: node
+    labels:
+      - key: logging-infra-fluentd
+        value: 'true'
+'''
```
Comment (tbielawa) [inline roles/lib_openshift/src/doc/label]: Some more examples expanding on those `state` choices would be very useful here.

File: roles/lib_openshift/src/test/integration/filter_plugins/filters.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+'''
+Custom filters for use in testing
+'''
+
+
+class FilterModule(object):
+    ''' Custom ansible filters '''
+
+    @staticmethod
+    def label_dict_to_key_value_list(label_dict):
```
Comment (tbielawa) [inline roles/lib_openshift/src/test/integration/filter_plugins/filters.py]: This seems to only be used by the integration tests. Should be add a disclaimer or other annotation in here to indicate this to possible contributors?

File: roles/lib_openshift/src/doc/label (hunk 2 above)
Comment (kwoodson) [inline roles/lib_openshift/src/doc/label]: Will fix.

File: roles/lib_openshift/src/doc/label (hunk 3 above)
Comment (kwoodson) [inline roles/lib_openshift/src/doc/label]: I will give an example.  each has key: <key>, value: <value> In order to key off specific keys or values.

File: roles/lib_openshift/src/test/integration/filter_plugins/filters.py (hunk 5 above)
Comment (kwoodson) [inline roles/lib_openshift/src/test/integration/filter_plugins/filters.py]: Sure, that sounds good.

File: roles/lib_openshift/src/doc/label (hunk 1 above)
Comment (kwoodson) [inline roles/lib_openshift/src/doc/label]: @tbielawa,

This is a good observation.  The 'present' state in declarative means create or update.  Think crud.
create|update = present
delete = absent
read = list
Basically, make-it-so and the 'so' being the state I'm passing in.  For the labeling module, make the labels present on the object.  Note that we did go outside the general pattern here and included an 'add'.  This just adds a label if needed.  That way you don't have to pass _ALL_ labels each time.  
If they already exist, exit.  
If they don't exist, create them.
If they aren't exact, update them.

I understand that the ansible terms for these can be less than ideal but its the same for the base modules.  Take a look at the user module's state:
absent|present - Whether the account should exist or not, taking action if the state is different from what is stated.

Hopefully this clears it up.  I'd be happy to discuss further.

File: roles/lib_openshift/src/doc/label (hunk 2 above)
Comment (kwoodson) [inline roles/lib_openshift/src/doc/label]: fixed to align with doc.  defaults to 'node'

**Comment by kwoodson** [comment]
@tbielawa,  Please approve when you feel its ready.

**Comment by tbielawa** [comment]
aos-ci-test

**Pull request merged**